pallets/api/src/messaging/benchmarking.rs

```diff
@@ -60,5 +60,80 @@ mod messaging_benchmarks {
 		)
 	}
 
+	#[benchmark]
+	fn xcm_new_query() {
+		let owner: AccountIdOf<T> = account("Alice", 0, SEED);
+		let message_id: [u8; 32] = [0; 32];
+		let responder = Location { parents: 1, interior: Junctions::Here };
+		let timeout = <BlockNumberOf<T> as One>::one() + frame_system::Pallet::<T>::block_number();
+		let callback =
+			Callback { selector: [0; 4], weight: 100.into(), spare_weight_creditor: owner.clone() };
+
+		pallet_balances::Pallet::<T>::make_free_balance_be(&owner, u32::MAX.into());
+
+		#[extrinsic_call]
+		Pallet::<T>::xcm_new_query(
+			RawOrigin::Signed(owner.clone()),
+			message_id.clone(),
+			responder.clone(),
+			timeout,
+			Some(callback.clone()),
+		);
+
+		assert_has_event::<T>(
+			crate::messaging::Event::XcmQueryCreated {
+				origin: owner,
+				id: message_id,
+				query_id: 0,
+				callback: Some(callback),
+			}
+			.into(),
+		)
+	}
+
+	/// x: Wether a successfully executing callback is provided.
+	#[benchmark]
+	fn xcm_response(x: Linear<0, 1>) {
+		let owner: AccountIdOf<T> = account("Alice", 0, SEED);
+		let message_id: [u8; 32] = [0; 32];
+		let responder = Location { parents: 1, interior: Junctions::Here };
+		let timeout = <BlockNumberOf<T> as One>::one() + frame_system::Pallet::<T>::block_number();
+		let callback = None;
+		let response = Response::ExecutionResult(None);
+		if x == 1 {
```

Comment:
A comment here would be great saying that the mock config has a always successful callback.